src/parser.ts

```diff
@@ -953,7 +954,8 @@ export class Parser extends DiagnosticEmitter {
         );
         typeParameter.defaultType = null;
       }
-      typeParameters.push(<TypeParameterNode>typeParameter);
+      if (!typeParameters) typeParameters = [<TypeParameterNode>typeParameter];
```

Comment:
Looks like the cast isn't necessary due to checking `if (!typeParameter) return null;` above

Reply:
Right! improved this